federation/pkg/federation-controller/ingress/ingress_controller.go

```diff
@@ -647,13 +648,15 @@ func (ic *IngressController) getMasterCluster() (master *federationapi.Cluster,
   If there is no master cluster, then fallbackUID is used (and hence this cluster becomes the master).
 */
 func (ic *IngressController) updateClusterIngressUIDToMasters(cluster *federationapi.Cluster, fallbackUID string) (string, error) {
-	masterCluster, masterUID, err := ic.getMasterCluster()
 	clusterObj, clusterErr := api.Scheme.DeepCopy(cluster) // Make a clone so that we don't clobber our input param
 	cluster, ok := clusterObj.(*federationapi.Cluster)
 	if clusterErr != nil || !ok {
-		glog.Errorf("Internal error: Failed clone cluster resource while attempting to add master ingress UID annotation (%q = %q) from master cluster %q to cluster %q, will try again later: %v", uidAnnotationKey, masterUID, masterCluster.Name, cluster.Name, err)
-		return "", err
+		errmsg := fmt.Sprintf("Internal error: Failed clone cluster %q resource: %v", cluster.Name, clusterErr)
```

Comment:
Yes, you are right, it's useful for debugging. And I am so sorry for my patch's description and removing detail log. I will update it.
I just think the following situation will be bug.
If the 'err' of 'masterCluster, masterUID, err := ic.getMasterCluster()' is nil and the 'clusterErr' of 'clusterObj, clusterErr := api.Scheme.DeepCopy(cluster)' is not nil, updateClusterIngressUIDToMasters() will return ("", nil).